lib/tests/streamlit/state/session_state_test.py

```diff
@@ -51,37 +61,47 @@ def test_make_state_old(self):
         assert self.session_state._old_state == {"foo": "bar", "baz": "qux"}
 
     def test_merged_state(self):
-        assert self.session_state._merged_state == {"foo": "bar", "baz": "qux"}
+        assert self.session_state._merged_state == {
+            "foo": "bar",
+            "foo2": "bar2",
+            "baz": "qux",
+        }
 
     def test_merged_state_new_state_priority(self):
         session_state = SessionState(
-            old_state={"foo": "bar"},
+            old_state={"foo": "bar", "foo2": "bar1"},
             new_state={"foo": "baz"},
         )
-        assert session_state._merged_state == {"foo": "baz"}
+        assert session_state._merged_state == {"foo": "baz", "foo2": "bar2"}
+
+        session_state = SessionState(old_state={}, new_state={"foo2": "baz"})
```

Comment:
I can't say I love creating the session state by setting internal attributes this way, but I don't know that doing the same setup using `make_state_old` would be much less coupled to the implementation.

Reply:
Yeah, neither feels great to me, but in this case I think that setting internal attributes is slightly preferable since using `make_state_old` and setting keys in `session_state` to write this test would make this test implicitly test/rely on the correctness of `make_state_old` and `__setitem__`, which are covered by their own unit tests